ccl/multiregionccl: TestIndexCleanupAfterAlterFromRegionalByRow failed

[ccl/multiregionccl.TestIndexCleanupAfterAlterFromRegionalByRow failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864626&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864626&tab=artifacts#/) on [master @ e1e10f99362e67771e00346fb0e006627eb67b09](https://github.com/cockroachdb/cockroach/commits/e1e10f99362e67771e00346fb0e006627eb67b09):


```
=== RUN   TestIndexCleanupAfterAlterFromRegionalByRow
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestIndexCleanupAfterAlterFromRegionalByRow168502644
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestIndexCleanupAfterAlterFromRegionalByRow
    regional_by_row_test.go:677: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestIndexCleanupAfterAlterFromRegionalByRow168502644
--- FAIL: TestIndexCleanupAfterAlterFromRegionalByRow (87.33s)
=== RUN   TestIndexCleanupAfterAlterFromRegionalByRow/REGIONAL_BY_TABLE
E210409 05:20:52.090494 275390 jobs/adopt.go:260  [n1] 1  job 648397606399868929: adoption completed with error attempting to GC indexes: clearing index 2: batch timestamp 1617945647.114023715,0 must be after replica GC threshold 1617945651.081808603,0
    regional_by_row_test.go:672: condition failed to evaluate within 45s: expected 4 jobs with status "succeeded", found 3. Jobs found: [{succeeded {"indexes": [{"dropTime": "1617945646092282058", "indexId": 1}], "parentId": 55}} {succeeded {"indexes": [{"dropTime": "1617945646307362686", "indexId": 3}], "parentId": 55}} {succeeded {"indexes": [{"dropTime": "1617945646308674889", "indexId": 4}], "parentId": 55}}]
        goroutine 266868 [running]:
        runtime/debug.Stack(0xc002495f30, 0x5a881a0, 0xc00502ca00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5c73f00, 0xc009dbe180, 0xc002495f30)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x6d
        github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestIndexCleanupAfterAlterFromRegionalByRow.func1(0xc009dbe180)
        	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/regional_by_row_test.go:672 +0x5bc
        testing.tRunner(0xc009dbe180, 0xc00ef76970)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
E210409 05:21:31.850865 268322 jobs/registry.go:681  [-] 2  error expiring job sessions: expire-sessions: context canceled
E210409 05:21:31.850972 268322 jobs/registry.go:686  [-] 3  failed to serve pause and cancel requests: context canceled
    --- FAIL: TestIndexCleanupAfterAlterFromRegionalByRow/REGIONAL_BY_TABLE (47.68s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestIndexCleanupAfterAlterFromRegionalByRow PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexCleanupAfterAlterFromRegionalByRow.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

--- a/pkg/ccl/multiregionccl/BUILD.bazel
+++ b/pkg/ccl/multiregionccl/BUILD.bazel
@@ -39,6 +39,7 @@ go_test(
         "//pkg/jobs",
         "//pkg/keys",
         "//pkg/kv",
+        "//pkg/kv/kvserver",
         "//pkg/roachpb",
         "//pkg/security",
         "//pkg/security/securitytest",

--- a/pkg/ccl/multiregionccl/regional_by_row_test.go
+++ b/pkg/ccl/multiregionccl/regional_by_row_test.go
@@ -20,6 +20,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv"
@@ -575,7 +576,20 @@ func TestIndexCleanupAfterAlterFromRegionalByRow(t *testing.T) {
 	} {
 		t.Run(tc.locality, func(t *testing.T) {
 			_, sqlDB, cleanup := multiregionccltestutils.TestingCreateMultiRegionCluster(
-				t, 3 /* numServers */, base.TestingKnobs{}, nil, /* baseDir */
+				t, 3 /* numServers */, base.TestingKnobs{
+					Store: &kvserver.StoreTestingKnobs{
+						// Disable the merge queue because it makes this test flakey
+						// under stress. Consider changing this when admin commands are
+						// better synchronized leading to less thrashing of the range cache.
+						// We may also need to retry ClearRange operations which bump into
+						// the GC threshold. Generally that's not a concern because
+						// generally that threshold isn't super small. The problem with
+						// ranges is that when they merge, as may happen during the parallel
+						// execution of a big ClearRange is that the highest threshold will
+						// be inherited.
+						DisableMergeQueue: true,
+					},
+				}, nil, /* baseDir */
 			)
 			defer cleanup()
 

